src/metabase/models/params/custom_values.clj

```diff
@@ -20,26 +20,29 @@
 
 ;;; ------------------------------------------------- source=static-list --------------------------------------------------
 (defn- query-matches
-  "Filter the values according to the `search-term`.
+  "Filters for values that match `query`.
 
   Values could have 2 shapes
-  - [value1, value2]
-  - [[value1, label1], [value2, label2]] - we search using label in this case"
+  - [[value1], [value2]]
+  - [[value1, label2], [value2, label2]] - we search using label in this case"
   [query values]
   (let [normalized-query (search/normalize query)]
-    (filter #(str/includes? (search/normalize (if (string? %)
-                                                %
-                                                ;; search by label
-                                                (second %)))
-                            normalized-query) values)))
+    (filter (fn [v]
+              (str/includes? (search/normalize (if (= (count v) 1)
+                                                 v
+                                                 (second v)))
+                             normalized-query))
+            values)))
 
 (defn- static-list-values
+  "Converts "
   [{values-source-options :values_source_config :as _param} query]
   (when-let [values (:values values-source-options)]
-    {:values          (if query
-                        (query-matches query values)
-                        values)
-     :has_more_values false}))
+    (let [values (map (fn [v] (if (string? v) [v] v)) values)]
```

Comment:
```suggestion
    (let [values (map (fn [v] (if (sequential? v) v [v])) values)]
```

I don't fully remember but could `v` be a number ?

either way using `sequential?` is safer I think.